main/api/src/mill/api/Result.scala

```diff
@@ -27,11 +27,14 @@ object Result {
    * @param value The value computed by the task.
    * @tparam T The result type of the computed task.
    */
-  case class Success[+T](value: T) extends Result[T] {
-    def map[V](f: T => V): Success[V] = Result.Success(f(value))
+  case class Success[+T](value: T, recalc: () => T) extends Result[T] {
```

Comment:
Can you elaborate on the purpose of `recalc` and add some API doc?

I only have a fuzzy idea how I should fill the `recalc` arg of a newly created `Success`.